### PR TITLE
Update troubleshooting chapter in User's Guide for poi code

### LINKED ISSUE #1211: Add documentation on "point of interest" code once it is incorporated
Emerging from #1208: Once we have added the "point of interest" code to the source tree (#1210), we'd like:

- [ ] Some documentation in the User's guide (troubleshooting section) about how to customize it and apply it in the code

- [ ] Some mention of this somewhere on the wiki, too?

Blocked by #1210

### PR DESCRIPTION
**Pull request by olyson**
### Description of changes
Add general tips on troubleshooting and instructions for using point_of_interest (poi) code.

### Specific notes

I added some general tips on troubleshooting.  I also pointed to the CIME User's Guide on troubleshooting which actually contains some of the tips that I use myself.  I also added documentation on using the point_of_interest code introduced by Bill Sacks.  There already were some sections on troubleshooting.  I don't really use those myself but I left those in just in case someone does.

Contributors other than yourself, if any:  Bill Sacks

CTSM Issues Fixed (include github issue #):
Resolves #1211 

Are answers expected to change (and if so in what way)? No, documentation only.

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any:
Verified proper ctsm-docs build on windows platform.

### CODE REVIEW

File: doc/source/users_guide/trouble-shooting/trouble-shooting.rst [hunk 1]
```diff
@@ -6,44 +6,121 @@
 Trouble Shooting 
 *********************
 
-In this chapter we give some guidance on what to do when you encounter some of the most common problems. We can't cover all the problems that a user could potentially have, but we will try to help you recognize some of the most common situations. And we'll give you some suggestions on how to approach the problem to come up with a solution.
+In this chapter we give some guidance on what to do when you encounter some of the most common problems.
 
-In general you will run into one of three type of problems:
```
Comment (billsacks) [inline doc/source/users_guide/trouble-shooting/trouble-shooting.rst:11]: Oh, I like the old wording: you WILL run into a problem :-)  (just kidding)

File: doc/source/users_guide/trouble-shooting/trouble-shooting.rst [hunk 2]
```diff
@@ -6,44 +6,121 @@
...
-
+Start with the `CIME Trouble Shooting Guide <http://esmci.github.io/cime/versions/master/html/users_guide/troubleshooting.html>`_ , especially if you encounter one of the first three types of problems. The CIME troubleshooting guide also provides some useful tips regarding run-time errors. If this doesn't identify and solve your problem, then try some of the suggestions below for run-time errors.
 
 General Advice on Debugging Run time Problems
 =============================================
 
-Here are some suggestions on how to track down a problem while running. In general if the problem still occurs for a simpler case, it will be easier to track down.
+The model has been run for thousands and thousands of simulation years in many different configurations, both fully-coupled and in land-only modes, without problems.  If you have modified the model in any way, by using either different input datasets or new or modified code, then obviously that is the first place to look if you encounter an error. However, the model is not completely infallible, as noted below.
```
Comment (billsacks) [inline doc/source/users_guide/trouble-shooting/trouble-shooting.rst:23]: I would delete "obviously", since that can sound condescending.

**Review by billsacks — CHANGES_REQUESTED** [review]
This is really helpful, @olyson - thanks a lot for adding this. It's really nice to have your process documented, and I like your use of a specific example.

I have just one tiny suggestion - see below.

File: doc/source/users_guide/trouble-shooting/trouble-shooting.rst (hunk 2 above)
Comment (olyson) [reply doc/source/users_guide/trouble-shooting/trouble-shooting.rst]: Done

File: doc/source/users_guide/trouble-shooting/trouble-shooting.rst (hunk 1 above)
Comment (olyson) [reply doc/source/users_guide/trouble-shooting/trouble-shooting.rst:11]: Ha! Perhaps: "The odds are good that you will run into a problem" :-)

**Review by billsacks — APPROVED** [review]
Thanks again!

**Pull request merged**